lms/djangoapps/course_home_api/outline/v1/views.py

```diff
@@ -152,11 +173,32 @@ def get(self, request, *args, **kwargs):
         transformers = BlockStructureTransformers()
         transformers += get_course_block_access_transformers(request.user)
         transformers += [
-            BlocksAPITransformer(None, None, depth=3),
+            BlocksAPITransformer(None, None, depth=4),
         ]
 
         course_blocks = get_course_blocks(request.user, course_usage_key, transformers, include_completion=True)
 
+        has_visited_course = False
+        resume_course_url = None
+        for block_key in course_blocks:
+            if course_blocks.get_xblock_field(block_key, 'category') == 'course':
+                if course_blocks.get_xblock_field(block_key, 'resume_block', False):
```

Comment:
This is bleeding knowledge a little bit. Now the caller of `get_resume_block` needs to know how resume blocks are defined in order to know whether it should actually call `get_resume_block`. You might consider moving this check inside the method, and rewriting the caller to avoid any resume block knowledge at all:

```
resume_block_id = self._get_resume_block(course_blocks, block_key)
if resume_block_id:
    has_visited_course = True
else:
    resume_block_id = self._get_start_block(course_blocks, block_key)
```

But since this are all internal methods, and not a public API, it doesn't really matter. I'm just finding nits.